src/gui/qgscolorrampbutton.cpp

```diff
@@ -36,7 +36,7 @@
 #include <QPushButton>
 #include <QWidget>
 
-QgsColorRampButton::QgsColorRampButton( QWidget *parent, const QString &dialogTitle )
+QgsColorRampButton::QgsColorRampButton( QWidget *parent SIP_TRANSFERTHIS, const QString &dialogTitle )
```

Comment:
@nyalldawson , I've added SIP_TRANSFERTHIS to the .cpp files, but the warnings are still there.